src/dramatiq_abort/middleware.py

```diff
@@ -126,22 +139,32 @@ def after_process_message(
 
     after_skip_message = after_process_message
 
-    def abort(self, message_id: str, abort_ttl: Optional[int] = None) -> None:
+    def abort(
+        self,
+        message_id: str,
+        mode: AbortMode = AbortMode.ABORT,
+        abort_ttl: Optional[int] = None,
+    ) -> None:
         if abort_ttl is None:
             abort_ttl = self.abort_ttl
-
-        self.backend.notify(self.id_to_key(message_id), ttl=abort_ttl)
+        modes = [AbortMode.CANCEL]
+        if mode is AbortMode.ABORT:
+            modes.append(AbortMode.ABORT)
+        keys = [self.id_to_key(message_id, mode) for mode in modes]
+        self.backend.notify(keys, ttl=abort_ttl)
 
     def _handle(self) -> None:
         message_ids = list(self.manager.abortables.keys())
         if not message_ids:
             time.sleep(self.wait_timeout / 1000)
             return
 
-        abort_keys = [self.id_to_key(id_) for id_ in message_ids]
+        abort_keys = [self.id_to_key(id_, AbortMode.ABORT) for id_ in message_ids]
         key = self.backend.wait_many(abort_keys, self.wait_timeout)
         if not key:
             return
+        else:
+            self.backend.poll(key)
```

Comment:
Why is this needed? `poll` has the same side-effects than `wait_many`

Reply:
Sorry, It's the omitted unused code during the recent review.